Image plugin add python plugin as a dependency even if we do not compile the plugin with python support

This issue was first discovered by @rosaliepl  : https://www.sofa-framework.org/community/forum/topic/error-with-cimg-plugin-with-dependancy-to-skeleton-h-does-not-compile/#post-8397

The image plugin supports the python plugin, but does not require it. However, when linking the image plugin into an external project by using the install path of sofa, the image plugin lists the python plugin as a required dependency. CMake will then return the following error:


```
CMake Error at CMakeLists.txt:50 (find_package):
  Found package configuration file:

/.../sofa/build/install/lib/cmake/image/imageConfig.cmake

  but it set image_FOUND to FALSE so package "image" is considered to be NOT
  FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing: SofaPython

-- Configuring incomplete, errors occurred!
```

To reproduce the bug
1. Compile SOFA with the image plugin activated, but without python plugin
2. Lauch cmake on an external project and add the image plugin as a dependency
`find_package(image REQUIRED)`

--- a/applications/plugins/image/CMakeLists.txt
+++ b/applications/plugins/image/CMakeLists.txt
@@ -89,8 +89,10 @@ find_package(CImg REQUIRED)
 configure_file(config.h.in "${CMAKE_BINARY_DIR}/include/image/config.h")
 install(FILES "${CMAKE_BINARY_DIR}/include/image/config.h" DESTINATION "include/image")
 
-# Config files and install rules for pythons scripts
-sofa_set_python_directory(${PROJECT_NAME} "python")
+if( SofaPython_FOUND )
+    # Config files and install rules for pythons scripts
+    sofa_set_python_directory(${PROJECT_NAME} "python")
+endif()
 
 if(NOT ${SOFA_NO_OPENGL})
     list(APPEND HEADER_FILES ImageViewer.h)

